content/6_Advanced/SPneg.mdx

```diff
@@ -215,6 +215,91 @@ int main() {
 
 </CPPSection>
 
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class CyclesFinding {
+	static final int N = 2501;
+	static int n, m;
+	static int[] parent = new int[N];
+	static long[] dist = new long[N];
+	static List<Edge> graph = new ArrayList<>();
+	static Kattio io = new Kattio();
+
+	// CodeSnip{Edge}
+	static class Edge {
+		int from;
+		int to;
+		long weight;
+
+		public Edge(int from, int to, long weight) {
+			this.from = from;
+			this.to = to;
+			this.weight = weight;
+		}
+	}
+	// EndCodeSnip
+
+	public static void main(String[] args) {
+		n = io.nextInt();
+		m = io.nextInt();
+		for (int i = 1; i <= m; i++) {
+			int a = io.nextInt();
+			int b = io.nextInt();
+			long c = (long)io.nextInt();
+			graph.add(new Edge(a, b, c));
+		}
+		bellmanFord(1);
+	}
+
+	private static void bellmanFord(int source) {
```

Comment:
the void here is kinda sus
imo it should return the cycle if there is one, and null if there isn't
`main` should then check for it and output accordingly
makes the flow of the program much better